crates/rpc/rpc/src/admin.rs

```diff
@@ -63,33 +63,43 @@ where
         Ok(true)
     }
 
+    /// Handler for `admin_peers`
     async fn peers(&self) -> RpcResult<Vec<PeerInfo>> {
         let peers = self.network.get_all_peers().await.to_rpc_result()?;
-        let peers = peers
-            .into_iter()
-            .map(|peer| PeerInfo {
-                id: Some(peer.remote_id.to_string()),
-                name: peer.client_version.to_string(),
-                caps: peer.capabilities.capabilities().iter().map(|cap| cap.to_string()).collect(),
-                network: PeerNetworkInfo {
-                    remote_address: peer.remote_addr.to_string(),
-                    local_address: peer
-                        .local_addr
-                        .unwrap_or_else(|| self.network.local_addr())
-                        .to_string(),
-                },
-                protocols: PeerProtocolsInfo {
-                    eth: Some(PeerEthProtocolInfo {
-                        difficulty: Some(peer.status.total_difficulty),
-                        head: peer.status.blockhash.to_string(),
-                        version: peer.status.version as u32,
-                    }),
-                    pip: None,
-                },
-            })
-            .collect();
+        let mut infos = Vec::with_capacity(peers.len());
 
-        Ok(peers)
+        for peer in peers {
+            if let Ok(pk) = id2pk(peer.remote_id) {
+                infos.push(PeerInfo {
+                    id: pk.to_string(),
```

Comment:
is this actually the public key?

Reply:
yes, it is, ref https://github.com/ethereum/go-ethereum/blob/7cf6a63687da2d9300fec495ff4bc978c4b70b65/p2p/enode/urlv4.go#L198-L203